LPS25H.class.nut

```diff
@@ -56,41 +65,75 @@ class LPS25H {
     }
 
     // -------------------------------------------------------------------------
-    function _readI2C(reg, numBytes) {
+    function _readReg(reg, numBytes) {
         local result = _i2c.read(_addr, reg.tochar(), numBytes);
         if (result == null) {
-            server.error("I2C read error: " + _i2c.readerror());
+            throw "I2C read error: " + _i2c.readerror();
         }
         return result;
     }
 
     // -------------------------------------------------------------------------
-    function _writeI2C(reg, ...) {
+    function _writeReg(reg, ...) {
         local s = reg.tochar();
         foreach (b in vargv) {
             s += b.tochar();
         }
         local result = _i2c.write(_addr, s);
         if (result) {
-            server.error("I2C write error: " + result);
+            throw "I2C write error: " + result;
         }
         return result;
```

Comment:
Agreed - can probably remove the return as it's guaranteed to be successful if no error was thrown.